numba/core/transforms.py

```diff
@@ -477,73 +476,126 @@ def _cfg_nodes_in_region(cfg, region_begin, region_end):
     return region_nodes
 
 
-def _legalize_withs_cfg(withs, cfg, blocks):
-    """Verify the CFG of the with-context(s).
-    """
-    doms = cfg.dominators()
-    postdoms = cfg.post_dominators()
-
-    # Verify that the with-context has no side-exits
-    for s, e in withs:
-        loc = blocks[s].loc
-        if s not in doms[e]:
-            # Not sure what condition can trigger this error.
-            msg = "Entry of with-context not dominating the exit."
-            raise errors.CompilerError(msg, loc=loc)
-        if e not in postdoms[s]:
-            msg = (
-                "Does not support with-context that contain branches "
-                "(i.e. break/return/raise) that can leave the with-context. "
-                "Details: exit of with-context not post-dominating the entry. "
-            )
-            raise errors.CompilerError(msg, loc=loc)
-
-
 def find_setupwiths(blocks):
     """Find all top-level with.
 
     Returns a list of ranges for the with-regions.
     """
     def find_ranges(blocks):
-        for blk in blocks.values():
-            for ew in blk.find_insts(ir.EnterWith):
-                if PYVERSION < (3, 9):
-                    end = ew.end
-                    for offset in blocks:
-                        if ew.end <= offset:
-                            end = offset
-                            break
-                else:
-                    # Since py3.9, the `with finally` handling is injected into
-                    # caller function. However, the numba byteflow doesn't
-                    # account for that block, which is where `ew.end` is
-                    # pointing to. We need to point to the block before
-                    # `ew.end`.
-                    end = ew.end
-                    last_offset = None
-                    for offset in blocks:
-                        if ew.end < offset:
-                            end = last_offset
-                            break
-                        last_offset = offset
-                yield ew.begin, end
-
-    def previously_occurred(start, known_ranges):
+
+        def is_setup_with(stmt):
+            return isinstance(stmt, ir.EnterWith)
+
+
+        def is_branch(stmt):
+            return isinstance(stmt, ir.Branch)
+
+
+        def is_terminator(stmt):
+            return isinstance(stmt, ir.Terminator)
```

Comment:
Given the use of this function later to determine which targets to add to the `to_visit` stack, is it really looking for **_any_** terminator, or just `Jump` and `Branch`? I suppose given those block terminators which terminate control flow, like `Raise`, have an empty target list it probably makes no difference.

Reply:
ok, let's leave it as-is.